File: app/src/main/java/tech/salroid/filmy/parser/MovieDetailsActivityParseWork.java
package tech.salroid.filmy.parser;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;
import java.util.ArrayList;
import java.util.List;
import tech.salroid.filmy.data.CastMemberDetailsData;
import tech.salroid.filmy.data.CrewMemberDetailsData;
import tech.salroid.filmy.data.SimilarMoviesData;

public class MovieDetailsActivityParseWork {

    private final String result;

    public MovieDetailsActivityParseWork(String result) {
        this.result = result;
    }

    public List<CastMemberDetailsData> parseCastMembers() {
        final List<CastMemberDetailsData> allCastMembers = new ArrayList<>();
        CastMemberDetailsData castMember;

        try {
            JSONObject jsonObject = new JSONObject(result);
            JSONArray jsonArray = jsonObject.getJSONArray("cast");

            for (int i = 0; i < jsonArray.length(); i++) {
                String id = jsonArray.getJSONObject(i).getString("id");
                String name = jsonArray.getJSONObject(i).getString("name");
                String rolePlayed = jsonArray.getJSONObject(i).getString("character");
                String displayProfile = jsonArray.getJSONObject(i).getString("profile_path");
                displayProfile = "http://image.tmdb.org/t/p/w185" + displayProfile;

                castMember = new CastMemberDetailsData(id);
                castMember.setCastId(id);
                castMember.setCastName(name);
                castMember.setCastRolePlayed(rolePlayed);
                castMember.setCastDisplayProfile(displayProfile);

                allCastMembers.add(castMember);
            }

        } catch (JSONException e) {
            e.printStackTrace();
        }

        return allCastMembers;
    }

    public List<CrewMemberDetailsData> parseCrewMembers() {
        final List<CrewMemberDetailsData> allCrewMembers = new ArrayList<CrewMemberDetailsData>();
        CrewMemberDetailsData crewMember;

        try {
            JSONObject jsonObject = new JSONObject(result);
            JSONArray crewArray = jsonObject.getJSONArray("crew");

            for (int i = 0; i < crewArray.length(); i++) {

                String memberId = crewArray.getJSONObject(i).getString("id");
                String memberJobDescription = crewArray.getJSONObject(i).getString("job");
                String memberName = crewArray.getJSONObject(i).getString("name");
                String memberProfile = crewArray.getJSONObject(i).getString("profile_path");
                memberProfile = "http://image.tmdb.org/t/p/w185" + memberProfile;

                crewMember = new CrewMemberDetailsData(memberId);

                if (!memberProfile.contains("null")) {
                    crewMember.setCrewMemberId(memberId);
                    crewMember.setCrewMemberName(memberName);
                    crewMember.setCrewMemberJob(memberJobDescription);
                    crewMember.setCrewMemberProfile(memberProfile);
                    allCrewMembers.add(crewMember);
                }
            }
        } catch (JSONException e) {
            e.printStackTrace();
        }
        return allCrewMembers;
    }

    public List<SimilarMoviesData> parse_similar_movies() {

        final List<SimilarMoviesData> similarArray = new ArrayList<>();
        SimilarMoviesData similar;

        try {
            JSONObject jsonObject = new JSONObject(result);
            JSONArray jsonArray = jsonObject.getJSONArray("results");

            for (int i = 0; i < jsonArray.length(); i++) {

                String id = (jsonArray.getJSONObject(i)).getString("id");
                String title = (jsonArray.getJSONObject(i)).getString("original_title");
                String poster = "http://image.tmdb.org/t/p/w185" + (jsonArray.getJSONObject(i)).getString("poster_path");

                similar = new SimilarMoviesData(id);

                if (!poster.contains("null")) {
                    similar.setId(id);
                    similar.setBanner(poster);
                    similar.setTitle(title);

                    similarArray.add(similar);
                }
            }

        } catch (JSONException e) {
            e.printStackTrace();
        }
        return similarArray;
    }
}

File: app/src/main/java/tech/salroid/filmy/parser/MovieDetailsActivityParseWork.kt
/*
package tech.salroid.filmy.parser

import org.json.JSONException
import org.json.JSONObject
import tech.salroid.filmy.data.Cast
import tech.salroid.filmy.data.Crew
import tech.salroid.filmy.data.SimilarMovie
import java.util.*

class MovieDetailsActivityParseWork(private val result: String) {

    fun parseCastMembers(): List<Cast> {
        val allCastMembers: MutableList<Cast> = ArrayList()
        var castMember: Cast

        try {
            val jsonObject = JSONObject(result)
            val jsonArray = jsonObject.getJSONArray("cast")

            for (i in 0 until jsonArray.length()) {
                val id = jsonArray.getJSONObject(i).getString("id")
                val name = jsonArray.getJSONObject(i).getString("name")
                val rolePlayed = jsonArray.getJSONObject(i).getString("character")
                var displayProfile = jsonArray.getJSONObject(i).getString("profile_path")
                displayProfile = "http://image.tmdb.org/t/p/w185$displayProfile"

                castMember = Cast(id)
                castMember.castId = id
                castMember.castName = name
                castMember.castRolePlayed = rolePlayed
                castMember.castDisplayProfile = displayProfile

                allCastMembers.add(castMember)
            }

        } catch (e: JSONException) {
            e.printStackTrace()
        }
        return allCastMembers
    }

    fun parseCrewMembers(): List<Crew> {
        val allCrewMembers: MutableList<Crew> = ArrayList()
        var crewMember: Crew

        try {
            val jsonObject = JSONObject(result)
            val crewArray = jsonObject.getJSONArray("crew")

            for (i in 0 until crewArray.length()) {
                val memberId = crewArray.getJSONObject(i).getString("id")
                val memberJobDescription = crewArray.getJSONObject(i).getString("job")
                val memberName = crewArray.getJSONObject(i).getString("name")
                var memberProfile = crewArray.getJSONObject(i).getString("profile_path")
                memberProfile = "http://image.tmdb.org/t/p/w185$memberProfile"
                crewMember = Crew(memberId)

                if (!memberProfile.contains("null")) {
                    crewMember.crewMemberId = memberId
                    crewMember.crewMemberName = memberName
                    crewMember.crewMemberJob = memberJobDescription
                    crewMember.crewMemberProfile = memberProfile
                    allCrewMembers.add(crewMember)
                }
            }

        } catch (e: JSONException) {
            e.printStackTrace()
        }
        return allCrewMembers
    }

    fun parseSimilarMovies(): List<SimilarMovie> {
        val similarArray: MutableList<SimilarMovie> = ArrayList()
        var similar: SimilarMovie

        try {
            val jsonObject = JSONObject(result)
            val jsonArray = jsonObject.getJSONArray("results")

            for (i in 0 until jsonArray.length()) {
                val id = jsonArray.getJSONObject(i).getString("id")
                val title = jsonArray.getJSONObject(i).getString("original_title")
                val poster = "http://image.tmdb.org/t/p/w185" + jsonArray.getJSONObject(i)
                    .getString("poster_path")

                similar = SimilarMovie(id)

                if (!poster.contains("null")) {
                    similar.id = id
                    similar.banner = poster
                    similar.title = title
                    similarArray.add(similar)
                }
            }
        } catch (e: JSONException) {
            e.printStackTrace()
        }
        return similarArray
    }
}*/
